SNIPS/snip-18.md

```diff
@@ -0,0 +1,159 @@
+* **Stake Delegation:** Delegators choose a Staker to whom they delegate their Stake, sharing rewards with said Staker. Stake Delegators do not need to run any software, as their chosen Staker runs it for them. This makes participation accessible to a broader audience, allowing users to participate in the Staking protocol without the technical and capital limitations.
+
+Both Stakers and Stake Delegators can unstake their funds, subject to protocol-defined latencies that ensure network stability and security. More details on these latencies can be found in the "Latencies" section.
+
+### Staking Rewards
+
+Staking rewards will be based on token minting. The rewards are calculated according to a minting curve, which follows Professor Noam Nisan’s [proposal](https://starkware.co/a-token-minting-proposal-to-manage-inflation/
+) (For more details on the minting curve, see the section below). Both Stakers and Stake Delegators earn rewards, depending on their respective stake and the reward-sharing constant (R) set by the Staker:
+
+* **Stakers:** Earn rewards proportional to their own stake and the total stake delegated to them, adjusted by the reward-sharing constant. The formula is:
+
+    `({self_stake} + {total_stake_delegated} * (1 - R)) * {rewards_constant} * {time_interval}`
+
+* **Stake Delegators:** Earn rewards proportional to their delegated stake and the reward-sharing constant. The formula is:
+
+    `{stake_delegated} * R * {rewards_constant} * {time_interval}`
+
+where `{rewards_constant}` is determined by the minting curve and depends on the total amount staked in the protocol. This reward structure incentivizes both staking, stake delegation, and competition among Stakers for delegators' stake.
+
+### Minting curve
+
+The minting curve addresses two critical aspects of the staking protocol by making the rewards dependent on the amount of STRK locked in the protocol. Here's how it works:
+
+* **Balancing Participation:** The more STRK tokens locked in the staking protocol, the lower the rewards become. This design ensures that not all STRK tokens are staked, allowing the Starknet community to continue using STRK for transaction fees and other activities. Additionally, it prevents unnecessary dilution of the token supply.
+
+* **Encouraging Staking:** When fewer STRK tokens are staked, the rewards get higher. This incentivizes participation in the staking protocol, guaranteeing a minimum level of economic security for the network by ensuring that a portion of STRK tokens is always staked.
+In the first stage, the minting curve will be based on Professor Noam Nisan's [proposal](https://starkware.co/a-token-minting-proposal-to-manage-inflation/), with a slight parameter variation. The minting curve is defined by the following formula:
+
+`M = C/10 * √ S`
```

Comment:
Can you give examples right below by injecting a few "scenario" value?

e.g.

C = 2.5 and S = 30%?
C = 4 and S = 50%?

by taking examples from Cosmos chains, Solana, and Eth L1?